syntax/printer_test.go

```diff
@@ -1059,6 +1059,18 @@ func TestPrintMinify(t *testing.T) {
 		},
 		samePrint("foo >bar 2>baz <etc"),
 		samePrint("<<-EOF\n$(a|b)\nEOF"),
+		{
+			"a=$(\n\tcat <<'EOF'\n  hello\nEOF\n)",
+			"a=$(cat \\\n<<'EOF'\n  hello\nEOF\n)",
+		},
+		{
+			"(\n\tcat <<EOF\n hello\nEOF\n)",
+			"(cat \\\n<<EOF\n hello\nEOF\n)",
+		},
+		{
+			"diff -y <(cat <<EOF\n1\n2\n3\nEOF\n) <(cat <<EOF\n1\n4\n3\nEOF\n)",
+			"diff -y <(cat \\\n<<EOF\n1\n2\n3\nEOF\n) <(\ncat <<EOF\n1\n4\n3\nEOF\n)",
```

Comment:
Something I realized while writing these examples is minified heredocs might not necessarily be minified! For example the following is 21 bytes:
```
a=$(cat <<EOF
EOF
)
```

And produces the following 22 bytes:
```
a=$(cat \
<<EOF
EOF
)
```

It's because of the `cat \`, but digging through the code I couldn't immediately see what causing that, or how to address it without breaking too much else. In the meantime, I figure valid shell is a lot better than invalid shell though!

Reply:
The `diff` example in the test is very funky, because the minified code has the first `cat` continuing its `<<EOF` onto the next line, but the next `cat` in the second proc subst is left alone as `cat <<EOF`.